### PR TITLE
[DO NOT MERGE] - Testing release notes link checker - WIP

### PR DESCRIPTION
**Pull request by rohennes**
Version(s):

Issue:

Link to docs preview:

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: scripts/check-RN-link-perms.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    # Iterate over the links and check their authorization status
+    for link in $links
+    do
+        response=$(curl -I "$link" 2>&1)
+
+        if echo "$response" | grep -q "permissionViolation"; then
+            echo "The link $link requires authentication."
+            protected_links+=("$link")
+        else
+            echo "The link $link does not require authentication."
```
Comment (aireilly) [inline scripts/check-RN-link-perms.sh:50]: IMO shouldn't need to flag links that are OK.

File: scripts/check-RN-link-perms.sh [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+# Get the repo path and files changed
+REPO_PATH="$(git rev-parse --show-toplevel)"
+FILES=$(git diff --name-only HEAD~1 HEAD --diff-filter=d "*.adoc" ':(exclude)_unused_topics/*')
+
+# Function to check links in updated release notes
+check_rn_links () {
+    
+    #Extract the release notes file from all the updated files
+    while IFS= read -r line; do
+        if echo "$line" | grep -q "release"; then
```
Comment (aireilly) [inline scripts/check-RN-link-perms.sh:14]: This seems brittle - there are lots of files with "release" in the filename.

File: scripts/check-RN-link-perms.sh (hunk 1 above)
Comment (rohennes) [reply scripts/check-RN-link-perms.sh:50]: Agreed.... just in there for testing purposes now

File: scripts/check-RN-link-perms.sh (hunk 2 above)
Comment (rohennes) [reply scripts/check-RN-link-perms.sh:14]: Good point. Perhaps "release-notes" would cover it

**Comment by rohennes** [comment]
testing complete, moving this to main

**Pull request closed without merging**